File: Harish Kumar/Dynamic Programming & Recursion/Longest Increasing Subsequence/Longest Increasing Subsequence.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        
        int dp[] = new int[nums.length];
        int maximum = 0;
        for(int i = 0; i< dp.length;i++)
        {
            int max = 0;
           for(int j = 0; j < i; j++)
           {
               if(nums[j] < nums[i])
               {
                   if(dp[j] > max )
                   {
                       max = dp[j];
                   }
               }
           }
            dp[i] = max+1;
            if(dp[i] > maximum)
            {
                maximum = dp[i];
            }
        }
        return maximum;
    }
}

File: Shreya/Recursion and DP/Longest Increasing Subsequence.cpp
class Solution
{
public:
    int lengthOfLIS(vector<int>& nums) {
        
        vector<int>vec(nums.size(),1);
        int maxm = 1;
        for(int i=1; i<nums.size(); i++)
        {
           for(int j=0; j<i; j++) 
           {
               if(nums[j]<nums[i])
               {
                   vec[i] = max(vec[i],vec[j]+1);
               }
               maxm = max(vec[i],maxm);
           }
        }
        return maxm;
    }
};
